redwoods-insurance-salesforce/force-app/main/default/lwc/incidentImageCarousel/__tests__/incidentImageCarousel.test.js

```diff
@@ -17,6 +30,47 @@ describe('c-incident-image-carousel', () => {
         document.body.appendChild(element);
 
         const labelElement = element.shadowRoot.querySelector('p');
-        expect(labelElement.textContent).toContain('no images');
+        expect(labelElement.textContent).toContain(
+            'There are currently no images of damage for this case.'
+        );
+    });
+
+    it('renders two pictures when the wire mock returns two', () => {
+        // Create initial element
+        const element = createElement('c-incident-image-carousel', {
+            is: IncidentImageCarousel
+        });
+        document.body.appendChild(element);
+
+        // Emit data from @wire
+        getRecordAdapter.emit('mockRecordId');
+        getRelatedPicturesAdapter.emit(mockTwoImages);
+
+        return Promise.resolve().then(() => {
+            const imageEls = element.shadowRoot.querySelectorAll(
+                'lightning-carousel-image'
+            );
+            expect(imageEls.length).toEqual(mockTwoImages.length);
+        });
+    });
+
+    it('renders no pictures when the wire adapter returns something other than an aray', () => {
+        // Create initial element
+        const element = createElement('c-incident-image-carousel', {
+            is: IncidentImageCarousel
+        });
+        document.body.appendChild(element);
+
+        // Emit data from @wire
+        getRecordAdapter.emit('mockRecordId');
+        // forcibly emit an object, instead of an array.
+        getRelatedPicturesAdapter.emit({});
```

Comment:
What would be the use case for this? I don't see any error handling in the component, nor an object return value from the Apex class that would lead to an empty object.

Reply:
It provides code coverage for the else statement of the wiredPictures function.

Reply:
In what cases would Apex return an `{}`? Looking more at the component itself I'm missing error handling, for when the wire adapter would throw an Apex exception.

Reply:
René this is a negative test. I'm intentionally emitting invalid data. No matter that the apex controller won't ever return such an empty object, the point is to test what the component does when it receives invalid data. To that point, it does not matter *what the invalid data is* so long as it receives invalid data. The If/Else in this component ( Lines 23-30) are testing for the data *files* being an Array. (line 23) Therefore, this test provides *invalid* data for that if condition - instead of an array, it gets an object. This forces the execution of the _else block_ on line 28-30. This test is valuable for   proving that when the If condition is handed something *other* than an Array, it properly nullifies the URL's array and nothing is shown to the user.